crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,930 @@
+/// Utilities for sorting constant lists of string literals.
+///
+/// Examples where these are useful:
+/// - Sorting `__all__` in the global scope,
+/// - Sorting `__slots__` or `__match_args__` in a class scope
+use std::borrow::Cow;
+use std::cmp::Ordering;
+
+use ruff_python_ast as ast;
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::{lexer, Mode, Tok, TokenKind};
+use ruff_python_stdlib::str::is_cased_uppercase;
+use ruff_python_trivia::leading_indentation;
+use ruff_source_file::Locator;
+use ruff_text_size::{Ranged, TextRange, TextSize};
+
+use is_macro;
+use natord;
+
+/// An enumeration of the different sorting styles
+/// currently supported for displays of string literals
+#[derive(Debug)]
```

Comment:
Nit: I think you should make this `Debug, Copy, Clone`. Then you can refer to it without references everywhere, e.g., you shouldn't need `const SORTING_STYLE: &SortingStyle = &SortingStyle::Natural;` -- you can just inline `SortingStyle::Natural` everywhere. If a struct is `Copy`, you don't need to call `.clone()` in order to pass an owned reference to it, since Rust will just copy it automatically. This struct is so tiny (smaller than a reference, even) that it's fine for it to be `Copy`.

Reply:
Oh of course, thanks!